files/aws_lifecycle_hooks/tools.py

```diff
@@ -74,19 +116,128 @@ def get_asg_name(
         instance_id: str,
         asg_client,
 ) -> str:
-    asg_info = asg_client.describe_auto_scaling_instances(InstanceIds=[instance_id])
-    asg_name = asg_info[u'AutoScalingInstances'][0][u'AutoScalingGroupName']
-    print("I am part of Auto Scaling Group {asg_name}".format(asg_name=asg_name))
+    """Use describe asg to get asg group name."""
+    import botocore.exceptions
```

Comment:
I don't like local imports. Is there a reason this can't be a module-wide import?

Reply:
I will leave this like this for now as this is a tools file that can be used in many different situations so i don't assume all dependencies will be installed on all systems using this lib.

Reply:
Yes, but moving this import to the top of this file will still do that... or am I missing something?

Reply:
No, because this method is only used in some specific setups...
F.I. On the Jenkins servers this is NOT used.

Reply:
OK, so you need it to be local because it will fail to load on e.g. Jenkins that doesn't have botocore installed?
If that is the case, please note this specific use-case in a comment